ros_cross_compile/data_collector.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Classes for time series data collection and writing said data to a file."""
+
+from datetime import datetime
+import json
+from pathlib import Path
+import time
+from typing import NamedTuple, Union
+
+
+class Datum(NamedTuple):
+    """Represents a collected piece of data."""
+
+    name: str
+    value: Union[int, float]
```

Comment:
can this just be float?

Reply:
The size of the docker image will be an int, right?

Reply:
It'll be a whole number but it may be simpler to report all data as floating point - the backend will receive it that way anyways, is my understanding.